pkg/pgmodel/querier/series_set.go

```diff
@@ -25,21 +24,38 @@ const (
 
 // pgxSeriesSet implements storage.SeriesSet.
 type pgxSeriesSet struct {
-	rowIdx  int
-	rows    []timescaleRow
-	err     error
-	querier labelQuerier
+	rowIdx     int
+	rows       []timescaleRow
+	labelIDMap map[int64]*labels.Label
+	err        error
+	querier    labelQuerier
 }
 
 // pgxSeriesSet must implement storage.SeriesSet
 var _ storage.SeriesSet = (*pgxSeriesSet)(nil)
 
-func buildSeriesSet(rows []timescaleRow, querier labelQuerier) storage.SeriesSet {
-	return &pgxSeriesSet{
-		rows:    rows,
-		querier: querier,
-		rowIdx:  -1,
+func buildSeriesSet(rows []timescaleRow, querier labelQuerier) (storage.SeriesSet, error) {
+	labelIDMap := make(map[int64]*labels.Label)
+	for i := range rows {
+		for _, id := range rows[i].labelIds {
+			if id == 0 {
+				continue
+			}
+			labelIDMap[id] = nil
+		}
+	}
+
+	err := querier.LabelsForIdMap(labelIDMap)
+	if err != nil {
+		return nil, err
```

Comment:
Why not return errorSeriesSet here and keep the same func signature?

Reply:
good point. fixed